### ISSUE TITLE
Useless use of ((…)) and $ on array indices undetected

### ISSUE THREAD

**Issue by Tamaranch** [koalaman/shellcheck#1666]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
declare -a a
i=0; j=1
a[(($i+$j))]=1
echo "${a[1]}"

```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

1. ^-- SC2004: $/${} is unnecessary on arithmetic variables.
2. ^-- SC????: ((…)) is unnecessary on array indices.

**Comment by koalaman** [comment]
SC2004 now triggers on array indices as well. This was an intentional omission going back to 2014 when ShellCheck struggled with associative arrays. Thanks!

**Comment by Tamaranch** [comment]
There is still no warning about the redundant arithmetic context though.

**Comment by koalaman** [comment]
These are redundant parentheses, not a redundant arithmetic context. However, you're definitely onto something: I found a number of instances of `array[$((x))]`, `array[(x)]`, and systematic use of `((..))` as parentheses inside arithmetic contexts presumably due to `$((..))`. These now have stylistic suggestions. Thanks!

Ideally it would also apply to `${array[..]}` but these are currently not parsed robustly enough to allow it.

**Comment by Tamaranch** [comment]
> These are redundant parentheses, not a redundant arithmetic context.

Oh yes, of course! Which explains why it was evaluated as an arithmetic expansion despite the absence of the `$` :joy:
Everything is well recognized for what it is now, thanks!
